zerver/models/prereg_users.py

```diff
@@ -73,6 +73,7 @@ class PreregistrationUser(models.Model):
     full_name = models.CharField(max_length=UserProfile.MAX_NAME_LENGTH, null=True)
     full_name_validated = models.BooleanField(default=False)
     referred_by = models.ForeignKey(UserProfile, null=True, on_delete=CASCADE)
+    send_notification = models.BooleanField(default=True)
```

Comment:
Let's call this `notify_referrer_on_join`; seems a lit more unambiguous what it means.

Also, do we want `False` to be the default in the database layer, since that's what value will be assigned to existing invitations?